File: pset2/vigenere/vigenere.c
#include <stdio.h>
#include <stdlib.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>

int main(int argc, string argv[])
{
    if (argc !=2){
        printf("Usage: ./vigenere k\n");
        return 1;
    }
    int len =strlen(argv[1]);
    int cipher[len];
    //make cipher of integers 0-25;
    for (int i=0;i<len;i++){
        if (!isalpha(argv[1][i])){
            printf("error\n");
            return 1;
        }
        if (isupper(argv[1][i])){
            cipher[i] = argv[1][i] - 65;
        }
        if (islower(argv[1][i])){
            cipher[i] = argv[1][i] - 97;
        }
    }
    //request plaintext from user
   printf("plaintext: ");
    string plaintext ;
    plaintext = get_string();
    printf("ciphertext: ");
    //initialize j as the cipher counter
    int j = 0;
    //iterate through plaintext
    for (int i=0, n=strlen(plaintext); i<n; i++){
        //non-alphabetical characters leave as-is, and don't increment j;
        if (!isalpha(plaintext[i])){
            printf("%c",plaintext[i]);
        }
        if (isupper(plaintext[i])){
            printf("%c",((plaintext[i]-65+cipher[j%len])%26)+65);
            j +=1;
        }
        if (islower(plaintext[i])){
            printf("%c",((plaintext[i]-97+cipher[j%len])%26)+97);
            j +=1;
        }
    }
   printf("\n");
}

File: pset6/vigenere.py
import cs50
import sys

if (len(sys.argv)) != 2:
    print("Usage vigenere.py k")
    exit(1)
if not sys.argv[1].isalpha():
    print("Key must be alphabetical")
    exit(1)
    
cipher = []
leng = len(sys.argv[1])
for i in range(leng):
    if sys.argv[1][i].isupper():
        cipher=cipher + [ord(sys.argv[1][i])-65]
    if sys.argv[1][i].islower():
        cipher=cipher + [ord(sys.argv[1][i])-97]

print("Plaintext: ",end="")
plain = cs50.get_string()
print("Ciphertext: ",end="")
j=0
for c in plain:
    if not c.isalpha():
        print(c,end="")
    if c.isupper():
        print("{}".format(chr((ord(c)-65+cipher[j%leng])%26+65)),end="")
        j+=1
    if c.islower():
        print("{}".format(chr((ord(c)-97+cipher[j%leng])%26+97)),end="")
        j+=1
print("")    
    